conans/test/functional/build_requires/test_build_require_invalid.py

```diff
@@ -0,0 +1,317 @@
+            from conans import ConanFile
+            from conans.errors import ConanInvalidConfiguration
+
+            class Conan(ConanFile):
+                name = "br"
+                version = "1.0.0"
+                settings = "os", "compiler", "build_type", "arch"
+
+                def build(self):
+                    if self.settings.build_type == "Debug":
+                        raise ConanInvalidConfiguration("br cannot be built in debug mode")
+
+                def package_id(self):
+                    del self.info.settings.compiler
+                    if self.settings.build_type == "Debug":
+                        compatible_pkg = self.info.clone()
+                        compatible_pkg.settings.build_type = "Release"
+                        self.compatible_packages.append(compatible_pkg)
+            """)
+    conanfile_consumer = textwrap.dedent("""
+        from conans import ConanFile
+
+        class Conan(ConanFile):
+            name = "libA"
+            version = "1.0.0"
+            settings = "os", "compiler", "build_type", "arch"
+            build_requires = "br/1.0.0"
+        """)
+    client.save({"br_validate.py": conanfile_validate,
+                 "br_configure.py": conanfile_configure,
+                 "br_build.py": conanfile_build,
+                 "br_validate_remove_build_type.py": conanfile_validate_remove_build_type,
+                 "br_configure_remove_build_type.py": conanfile_configure_remove_build_type,
+                 "br_build_remove_build_type.py": conanfile_build_remove_build_type,
+                 "br_validate_compatible.py": conanfile_validate_compatible,
+                 "br_configure_compatible.py": conanfile_configure_compatible,
+                 "br_build_compatible.py": conanfile_build_compatible,
+                 "consumer.py": conanfile_consumer})
+    return client
+
+
+def test_create_invalid_validate(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_invalid_configure(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_configure.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_invalid_build(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_build.py")
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
+
+
+def test_create_validate_remove_build_type(setup_client_with_build_requires):
+    client = setup_client_with_build_requires
+    client.run("create br_validate_remove_build_type.py")
+    # FIXME: This should NOT fail with -> br/1.0.0: Invalid ID: br cannot be built in debug mode
+    client.run("create consumer.py -s build_type=Debug", assert_error=True)
```

Comment:
This is the use-case we are exercising currently in ConanCenter and **we need to fix in the Conan client**. Because of this error we are not generating some configurations of libraries.

Reply:
Yes, agree, this is the main case I am trying to address in the client.